src/parser/mod.rs

```diff
@@ -9416,27 +9416,42 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_set_role(&mut self, modifier: Option<Keyword>) -> Option<Statement> {
+        let context_modifier = match modifier {
+            Some(Keyword::LOCAL) => ContextModifier::Local,
+            Some(Keyword::SESSION) => ContextModifier::Session,
+            _ => ContextModifier::None,
+        };
+
+        if self.parse_keyword(Keyword::NONE) {
+            Some(Statement::SetRole {
+                context_modifier,
+                role_name: None,
+            })
+        } else if matches!(
+            self.peek_token().token,
+            Token::Word(_) | Token::DoubleQuotedString(_) | Token::SingleQuotedString(_)
+        ) {
```

Comment:
Not sure I followed the intention with this change, it seems to match the behavior of `parse_identifier()`?